pennylane/ops/qubit.py

```diff
@@ -43,26 +44,27 @@ class Hadamard(Observable, Operation):
     num_params = 0
     num_wires = 1
     par_domain = None
-    eigvals = pauli_eigs(1)
+    eigvals = lambda self: pauli_eigs(1)
 
     @staticmethod
-    def matrix():
-        r"""Matrix representation of the Hadamard operator.
+    def _matrix(*params):
+        return np.array([[1, 1], [1, -1]]) / np.sqrt(2)
 
-        Returns:
-            array[complex]: matrix representation of the Hadamard operator
-        """
-        return np.array([[1, 1], [1, -1]], dtype=np.complex128) / np.sqrt(2)
+    def diagonalizing_gates(self):
+        r"""Rotates the specified wires such that they
+        are in the eigenbasis of the Hadamard operator.
 
-    @staticmethod
-    def diagonalizing_gates():
-        r"""Diagonalize Hadamard.
+        For the Hadamard operator,
+
+        .. math:: H = U^\dagger Z U
+
+        where :math:`U = R_y(-\pi/4)`.
 
         Returns:
-            list(qml.Operation): A list of gates that diagonalize Hadamard in
-            the computational basis.
+            list(qml.Operation): A list of gates that diagonalize PauliY in the
```

Comment:
```suggestion
            list(qml.Operation): A list of gates that diagonalize Hadamard in the
```

Reply:
``RY`` was used here, though it's for ``Hadamard`` :)

Reply:
Good catch!